### PR TITLE
create the validation engine by further specializing the existing validator services

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

File: validator/config.py [hunk 1]
```diff
@@ -8,31 +8,105 @@
...
-RDF_VALIDATOR_UI_LOCATION = os.environ.get('RDF_VALIDATOR_UI_LOCATION', 'http://fingerprinter-api')
-RDF_VALIDATOR_UI_PORT = os.environ.get('RDF_VALIDATOR_UI_PORT', 4010)
-RDF_VALIDATOR_UI_SERVICE = str(RDF_VALIDATOR_UI_LOCATION) + ":" + str(RDF_VALIDATOR_UI_PORT)
-RDF_VALIDATOR_UI_SECRET_KEY = os.environ.get('RDF_VALIDATOR_UI_SECRET_KEY', 'secret key api')
-
-RDF_VALIDATOR_LOGGER = 'validator'
+from typing import Optional
+
+
+class classproperty(property):
```
Comment (costezki) [inline validator/config.py:18]: I am not sure we need this.

Simply use `@property` decorator. That should suffice

File: validator/config.py [hunk 2]
```diff
@@ -8,31 +8,105 @@
...
+    def __get__(self, cls, owner):
+        return classmethod(self.fget).__get__(None, owner)()
+
+
+class ValidatorConfig:
+    logger_name = 'validator'
+    logger = logging.getLogger(logger_name)
+
+    @classproperty
+    def RDF_VALIDATOR_LOGGER(cls) -> str:
```
Comment (costezki) [inline validator/config.py:28]: @property?

File: validator/config.py [hunk 3]
```diff
@@ -8,31 +8,105 @@
...
+        value = cls.logger_name
+        cls.logger.debug(value)
+        return value
+
+    @classproperty
```
Comment (costezki) [inline validator/config.py:33]: @property?

File: validator/config.py [hunk 4]
```diff
@@ -8,31 +8,105 @@
...
+    def RDF_VALIDATOR_UI_SECRET_KEY(cls) -> str:
+        value = os.environ.get('RDF_VALIDATOR_UI_SECRET_KEY', 'secret key api')
+        cls.logger.debug(value)
+        return value
 
 
 class FlaskConfig:
```
Comment (costezki) [inline validator/config.py:112]: Now the questions is: could we try and unify also the FlaskConfig class into the ValidatorConfig?

File: validator/entrypoints/api/__init__.py [hunk 5]
```diff
@@ -11,10 +11,10 @@
 
 import connexion
 
-from validator.config import RDF_VALIDATOR_API_SECRET_KEY
+from validator.config import ValidatorConfig as config
```
Comment (costezki) [inline validator/entrypoints/api/__init__.py:14]: smooth!

File: validator/entrypoints/api/helpers.py [hunk 6]
```diff
@@ -9,8 +9,15 @@
 Helper methods
```
Comment (costezki) [inline validator/entrypoints/api/helpers.py:9]: Looks like another set of `validator` helpes. Idem as above,

File: validator/entrypoints/api/openapi/validator.yaml [hunk 7]
```diff
@@ -38,9 +38,15 @@ paths:
...
       responses:
         200:
-          description: Request to validate endpoint successfully accepted for processing.
+          description: Request to validate endpoint successfully accepted for processing
+        400:
+          description: This configuration of the validator doesn't accept external SHACL shapes
+        415:
+          description: Unacceptable file types
+        422:
+          description: Wront report extension format
```
Comment (costezki) [inline validator/entrypoints/api/openapi/validator.yaml:47]: `wrong`

File: validator/entrypoints/api/openapi/validator.yaml [hunk 8]
```diff
@@ -86,13 +92,18 @@ paths:
...
```
Comment (costezki) [inline validator/entrypoints/api/openapi/validator.yaml:104]: `wrong`

File: validator/entrypoints/api/run.py [hunk 9]
```diff
@@ -10,10 +10,10 @@
 """
 import logging
 
-from validator.config import RDF_VALIDATOR_DEBUG, ProductionConfig, DevelopmentConfig
+from validator.config import ProductionConfig, DevelopmentConfig, ValidatorConfig as config
 from validator.entrypoints.api import app
 
-if RDF_VALIDATOR_DEBUG:
+if config.RDF_VALIDATOR_DEBUG:
```
Comment (costezki) [inline validator/entrypoints/api/run.py:16]: smooth

File: validator/entrypoints/ui/templates/base.html [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE HTML>
 <html lang="en">
 <head>
-    <title>Meaningfy - RDF Differ</title>
+    <title>{{ validator_name | default('RDF Differ', true) }}</title>
```
Comment (costezki) [inline validator/entrypoints/ui/templates/base.html:4]: Default is `RDF Validator`

Comment (costezki) [reply validator/entrypoints/ui/templates/base.html:4]: but again, this default should also be handled in the config module

File: validator/entrypoints/ui/templates/base.html [hunk 11]
```diff
@@ -12,30 +12,31 @@
...
-{% block navbar %}
-   <nav class="navbar navbar-expand-lg navbar-light bg-light green lighten-3">
-        <div class="container">
-          <a href="{{ url_for('index') }}" style="color: #1e1e1e">RDF Validator</a>
+    {% set active_page = active_page|default('index') -%}
+    {% block navbar %}
+        <nav class="navbar navbar-expand-lg navbar-light bg-light green lighten-3">
+            <div class="container">
+                <a id="website-name" href="{{ url_for('index') }}"
+                   style="color: #1e1e1e">{{ validator_name | default('RDF Differ', true) }}</a>
```
Comment (costezki) [inline validator/entrypoints/ui/templates/base.html:25]: `RDF validator` not differ

File: validator/service_layer/handlers.py [hunk 12]
```diff
@@ -156,29 +120,26 @@ def generate_validation_report(path_to_report: Union[str, Path]) -> str:
...
-        json.dump(config, config_file)
+        json.dump(config_data, config_file)
         config_file.truncate()
 
 
-def create_file_name(filename: str, file_type: str = TTL_EXTENSION) -> str:
-    return Path(filename).stem + f'-report.{file_type}'
-
-
 def create_report(location: str, html_report: str, ttl_report: str, extension: str, file_name: str):
```
Comment (costezki) [inline validator/service_layer/handlers.py:143]: is this method named using a different logic than `build_report_from_sparql_endpoint` should it be called `build_report_from_file` ?

File: validator/service_layer/helpers.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+logger = logging.getLogger(config.RDF_VALIDATOR_LOGGER)
+
+
+class SHACLShapesMissing(Exception):
+    """
+        An exception to be raised when the lookup for the SHACL shape files fails.
+    """
+
+
+def create_file_name(filename: str, file_type: str = TTL_EXTENSION) -> str:
```
Comment (costezki) [inline validator/service_layer/helpers.py:25]: is this function necessary?

**Review by costezki — APPROVED** [review]
looking good

**Review by costezki — APPROVED** [review]

File: validator/config.py (hunk 1 above)
Comment (cosletmihai) [reply validator/config.py:18]: to use `@property` requires an instance (an object) of the object to run the method. the `@classproperty` gives the same functionality of `@property` **and** also works with `class` instances.

File: validator/entrypoints/api/__init__.py (hunk 5 above)
Comment (cosletmihai) [reply validator/entrypoints/api/__init__.py:14]: thank you

File: validator/entrypoints/api/handlers.py [hunk 14]
```diff
@@ -14,18 +14,17 @@
...
+from validator.entrypoints.api.helpers import DEFAULT_REPORT_EXTENSION, check_for_file_exceptions
 from validator.service_layer.handlers import build_report_from_file, build_report_from_sparql_endpoint
 
-logger = logging.getLogger(RDF_VALIDATOR_LOGGER)
+logger = logging.getLogger(config.RDF_VALIDATOR_LOGGER)
 
 
 def validate_file(data_file: FileStorage,
-                  schema_file0: FileStorage, schema_file1: FileStorage = None, schema_file2: FileStorage = None,
+                  schema_file0: FileStorage = None, schema_file1: FileStorage = None, schema_file2: FileStorage = None,
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply validator/entrypoints/api/handlers.py:27]: these methods are from `handlers.py` used by `connexion`, not `helpers.py`

File: validator/service_layer/handlers.py (hunk 12 above)
Comment (cosletmihai) [reply validator/service_layer/handlers.py:143]: this is a generic method that is independent of the type (from `file` or `endpoint`)

File: validator/service_layer/helpers.py (hunk 13 above)
Comment (cosletmihai) [reply validator/service_layer/helpers.py:25]: it's useful as it hides the tedious implementation

File: validator/config.py (hunk 1 above)
Comment (costezki) [reply validator/config.py:18]: I see no problem instantiating the class then. In general, I am in favour of using standard tools, and only if something really is missing or specific to us, then implement it.

File: validator/entrypoints/ui/templates/base.html (hunk 10 above)
Comment (costezki) [reply validator/entrypoints/ui/templates/base.html]: I believe 'RDF Differ' is the wrong default value. Change it pls as suggested in the previous comment.

Comment (cosletmihai) [reply validator/entrypoints/ui/templates/base.html]: I changed it in the last commit: https://github.com/meaningfy-ws/rdf-validator-ws/pull/44/commits/50291fd2ec82bad1ecc167f61feaaf8aa39e4647

File: validator/config.py (hunk 1 above)
Comment (cosletmihai) [reply validator/config.py:18]: I'll make the adjustment on lam and create a task for updating this

**Pull request merged**